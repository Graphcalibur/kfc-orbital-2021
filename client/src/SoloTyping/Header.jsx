import React from 'react';

const Header = (props) => {
    return <b className="text">
        Language: {props.language}
    </b>
}

export default Header;