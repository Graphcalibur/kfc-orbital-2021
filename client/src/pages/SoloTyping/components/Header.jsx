import React from 'react';
import Timer from "./Timer";

const Header = (props) => {
    return <div className="d-flex flex-sm-row gap-5">
        <b className="text">
            Language: {props.language}
        </b>

        <Timer elapsed_time={props.elapsed_time} typing={props.typing} />
    </div>
}

export default Header; 