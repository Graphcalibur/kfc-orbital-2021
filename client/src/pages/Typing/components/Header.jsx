import React from "react";
import { Button, Container, OverlayTrigger, Tooltip } from "react-bootstrap";

const Header = (props) => {
  const popover = (
    <Tooltip>
      Type the code in each line and hit Enter when you reach the end. Any
      errors made will have to be backspaced and fixed before continuing. Timing
      begins when
      {props.is_solo ? " you start typing " : " the countdown finishes "}
      and ends after the last line.
    </Tooltip>
  );

  return (
    <Container className="shadow p-3 ms-3 align-content-start box">
      <p className="text">
        <b>Language:</b> {props.language}
        <br />
        <b>Length (Characters):</b> {props.code_length}
        <br />
        <b>Length (Lines):</b> {props.code_lines}
      </p>

      <OverlayTrigger trigger="click" placement="bottom" overlay={popover}>
        <Button variant="outline-primary">How to Play</Button>
      </OverlayTrigger>
    </Container>
  );
};

export default Header;
