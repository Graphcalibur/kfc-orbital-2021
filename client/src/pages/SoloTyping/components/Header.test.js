import React from "react";
import Header from "./Header";
import { create } from "react-test-renderer";

describe("Testing Header component in page SoloTyping", () => {
  test("Test 1", () => {
    const tree = create(
      <Header language={"Python"} code_length={80} code_lines={7} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test("Test 2", () => {
    const tree = create(
      <Header language={"C++"} code_length={6931} code_lines={314} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
