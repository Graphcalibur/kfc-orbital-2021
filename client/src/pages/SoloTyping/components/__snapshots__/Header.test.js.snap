// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component in page SoloTyping Test 1 1`] = `
<div
  className="shadow p-3 ms-3 align-content-start box container"
>
  <p
    className="text"
  >
    <b>
      Language:
    </b>
     
    Python
    <br />
    <b>
      Length (Characters):
    </b>
     
    80
    <br />
    <b>
      Length (Lines):
    </b>
     
    7
  </p>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    How to Play
  </button>
</div>
`;

exports[`Testing Header component in page SoloTyping Test 2 1`] = `
<div
  className="shadow p-3 ms-3 align-content-start box container"
>
  <p
    className="text"
  >
    <b>
      Language:
    </b>
     
    C++
    <br />
    <b>
      Length (Characters):
    </b>
     
    6931
    <br />
    <b>
      Length (Lines):
    </b>
     
    314
  </p>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    How to Play
  </button>
</div>
`;
